packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -248,6 +248,7 @@ class WebpackCLI extends GroupHelper {
     async runOptionGroups() {
         await Promise.resolve()
             .then(() => this._handleGroupHelper(this.zeroConfigGroup))
+            .then(() => this._handleGroupHelper(this.basicGroup))
```

Comment:
Why executing the basic group twice?

Reply:
> I think the solution should be different. I left some feedback on how to tackle the problem. Let me know what you think

ok.

## Entry rules

to get the right entry, I should consider the different entry priority

- command entry exists, use the command entry
- no command entry, config entry exists, use the config entry
- no command entry, no config entry, use defaultEntry

The origin BasicGroup would get the defaultEntry or command entry, which can't be distinguished when executing webpackMerge, caused defaultEntry override config entry.
because BasicGoup execute after ConfigGroup

## Why execute BasicGroup twice?

The first time, to get defaultEntry, and BasicGroup execute before ConfigGoup, the config entry can override defaultEntry correctly.

The second time, to get command entry, and BasscGroup execute after ConfigGoup, the command entry can override config entry correctly.

## Why not using webpackMerge strategies

because entry may be string, while strategies only impact object or array merge. Maybe strategies can do but I don't know. if so, please let me know.

## improvement

The implemented method is not good, I am considering how to improve it, maybe I can pass the configuration to BasicGroup to decide to use defaultEntry or command entry. if you have a good idea, please let me know.

Thanks for reviewing.

Reply:
I had the issue while working with the `output` property. Try to apply the same strategy. Strategies work for all type of entries

Reply:
Thanks, I will try it in this way.